test/images/sample-device-plugin/sampledeviceplugin.go

```
@@ -0,0 +1,98 @@
+	dm "k8s.io/kubernetes/pkg/kubelet/cm/devicemanager"
+)
+
+const (
+	resourceName = "fake.com/resource"
+)
+
+// stubAllocFunc will pass to stub device plugin
+func stubAllocFunc(r *pluginapi.AllocateRequest, devs map[string]pluginapi.Device) (*pluginapi.AllocateResponse, error) {
+	var responses pluginapi.AllocateResponse
+	for _, req := range r.ContainerRequests {
+		response := &pluginapi.ContainerAllocateResponse{}
+		for _, requestID := range req.DevicesIDs {
+			dev, ok := devs[requestID]
+			if !ok {
+				return nil, fmt.Errorf("invalid allocation request with non-existing device %s", requestID)
+			}
+
+			if dev.Health != pluginapi.Healthy {
+				return nil, fmt.Errorf("invalid allocation request with unhealthy device: %s", requestID)
+			}
+
+			// create fake device file
+			fpath := filepath.Join("/tmp", dev.ID)
+
+			// clean first
+			if err := os.RemoveAll(fpath); err != nil {
+				return nil, fmt.Errorf("failed to clean fake device file from previous run: %s", err)
+			}
+
+			f, err := os.Create(fpath)
+			if err != nil && !os.IsExist(err) {
+				return nil, fmt.Errorf("failed to create fake device file: %s", err)
+			}
+
+			f.Close()
+
+			response.Mounts = append(response.Mounts, &pluginapi.Mount{
+				ContainerPath: fpath,
+				HostPath:      fpath,
+			})
+		}
+		responses.ContainerResponses = append(responses.ContainerResponses, response)
+	}
+
+	return &responses, nil
+}
+
+func main() {
+	devs := []*pluginapi.Device{
+		{ID: "Dev-1", Health: pluginapi.Healthy},
+		{ID: "Dev-2", Health: pluginapi.Healthy},
+	}
+
+	pluginSocksDir := os.Getenv("PLUGIN_SOCK_DIR")
+	klog.Infof("pluginSocksDir: %s", pluginSocksDir)
+	if pluginSocksDir == "" {
+		fmt.Println("Empty pluginSocksDir")
+		return
+	}
+	socketPath := pluginSocksDir + "/dp." + fmt.Sprintf("%d", time.Now().Unix())
+
+	dp1 := dm.NewDevicePluginStub(devs, socketPath, resourceName, false)
+	err := dp1.Start()
+	fmt.Println("Start error: ", err)
```

shouldn't we abort/exit if there is an error rather than continuing silently?

---

also please use `klog` rather than `fmt.Println`.